lib/options.nix

```
@@ -45,158 +47,262 @@ let
 in
 rec {
 
-  /* Returns true when the given argument is an option
+  /**
+    Returns true when the given argument is an option
 
-     Type: isOption :: a -> bool
+    # Inputs
+
+    `value`
+    : Any value to check whether it is an option
+
+    # Examples
+    :::{.example}
+    ## `lib.options.isOption` usage example
+
+    ```nix
+    isOption 1             // => false
+    isOption (mkOption {}) // => true
+    ```
+
+    :::
+
+    # Type
+
+    ```
+    isOption :: a -> bool
```

`bool` is also a module system type; and Nix types don't have a standardized syntax yet.
I'd suggest to differentiate them, because a _`types.bool` definition_ is not necessarily a valid boolean.

```suggestion
    isOption :: a -> Bool
```

This is also more consistent with the Haskell-inspired type syntax you're using.